File: Week 9/templates/E/E_frozenrose.py

import sys
adj = []
def frozenrose(n, c):
    # Input format: 'n' is the number of nodes, 'c' is the central node
    # 'adj' is a list of edges of length 'n + 1' (1-indexed, same as input)
    # 'adj' will be initialized so that adj[u] is a list of edges from node u
    # Each edge is a tuple (v, weight)
    # Return the minimum cost to ensure no water leaves from a sprinkler
    return -1

# A note about this problem; it reads until EOF, which means that if you're
# typing the input in manually, you need to press Ctrl+C after the last test case,
# or type "end" on a line by itself (in case it's not flushing the output)
for line in sys.stdin:
    if line.strip() == "end":
        break
    n, c = map(int, line.split())
    adj = [[] for _ in range(n + 1)]
    for _ in range(n - 1):
        u, v, weight = map(int, input().split())
        adj[u].append((v, weight))
        adj[v].append((u, weight))
    print(frozenrose(n,c))

File: Week 9/templates/E/E_frozenrose.java
import java.io.*;
import java.util.*;

class Main{
    // Make sure the class name is Main if you use this template!
    static class Edge{
        int v, weight;
        Edge(int v, int weight){
            this.v = v;
            this.weight = weight;
        }
    }
    static ArrayList<Edge>[] adj;
    static int frozenrose(int n, int c){
        // Input format: 'n' is the number of nodes, 'c' is the central node
        // 'adj' is a list of edges of length 'n + 1' (1-indexed, same as input)
        // 'adj' will be initialized so that adj[u] is a list of edges from node u
        // Return the minimum cost to ensure no water leaves from a sprinkler
        return -1;
    }
    public static void main(String[] args) throws Exception{
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        // A note about this problem; it reads until EOF, so if you type in the input manually,
        // you need paste the input all at once with this implementation, or put it in a file
        // and use the console pipeline to read from the file in stdin
        while(!in.ready()); // You could remove this line if you didn't want to ever paste in the input manually
        while(in.ready()){
            StringTokenizer st = new StringTokenizer(in.readLine());
            int n = Integer.parseInt(st.nextToken());
            int c = Integer.parseInt(st.nextToken());
            adj = new ArrayList[n + 1];
            for(int i = 1; i <= n; i++){
                adj[i] = new ArrayList<Edge>();
            }
            for(int i = 0; i < n - 1; i++){
                st = new StringTokenizer(in.readLine());
                int u = Integer.parseInt(st.nextToken());
                int v = Integer.parseInt(st.nextToken());
                int weight = Integer.parseInt(st.nextToken());
                adj[u].add(new Edge(v,weight));
                adj[v].add(new Edge(u,weight));
            }
            System.out.println(frozenrose(n,c));
        }
    }
}